runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -207,7 +211,9 @@ function STHighlighter:new(bufnr)
     buffer = self.bufnr,
     group = self.augroup,
     callback = function()
-      self:send_request()
+      if M.is_enabled({ bufnr = bufnr }) then
+        self:send_request()
+      end
```

Comment:
Here's another one. Seems like maybe it'd be find to just call send_request() and let that function handle checking the enabled flag.